src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala

```diff
@@ -9,90 +9,216 @@ import org.apache.spark.ml.param.DataFrameEquality
+        val langCode = result.getAs[String]("iso6391Name")
+        val confidence = result.getAs[Double]("confidenceScore")
+        val charactersCount = statistics.getAs[Int]("charactersCount")
+        val transactionsCount = statistics.getAs[Int]("transactionsCount")
+
+
+        assert(modelVersion == "2021-01-05")
+
+        if (statistics == "true") {
+          assert(charactersCount == 2)
+          assert(transactionsCount == 1)
+        }
+        if (statistics == "false") {
+          print("Statistics will not be shown.")
+        }
+      });
+    }
 
-    assert(replies(0).getString(0) == "English" && replies(2).getString(0) == "Spanish" )
-    assert(replies(3).getString(0) == "Chinese_Traditional")
-    assert(replies(0).getString(1) == "en" && replies(2).getString(1) == "es")
-
+    test("Language Detection - Print Schema") {
+      detector.transform(df).printSchema()
+      detector.transform(df).show()
+    }
   }
-}
 
-class TextSentimentSuiteV4 extends TestBase with DataFrameEquality with TextKey {
+  class TextSentimentSuiteV4 extends TestBase with DataFrameEquality with TextKey {
 
-  import spark.implicits._
+    import spark.implicits._
 
-  val options: Option[TextAnalyticsRequestOptions] = Some(new TextAnalyticsRequestOptions()
-    .setIncludeStatistics(true))
+    val options: Option[TextAnalyticsRequestOptions] = Some(new TextAnalyticsRequestOptions()
+      .setIncludeStatistics(true))
 
-  lazy val df: DataFrame = Seq(
-    "Hello world. This is some input text that I love.",
-    "I am sad",
-    "I am feeling okay"
-  ).toDF("text")
+    lazy val df: DataFrame = Seq(
+      "Hello world. This is some input text that I love.",
+      "I am sad",
+      "I am feeling okay"
+    ).toDF("In - Sentiment")
 
-  lazy val detector: TextSentimentV4 = new TextSentimentV4(options)
-    .setSubscriptionKey(textKey)
-    .setEndpoint("https://eastus.api.cognitive.microsoft.com/")
-    .setInputCol("text")
+    lazy val detector: TextSentimentV4 = new TextSentimentV4(options)
+      .setSubscriptionKey(textKey)
+      .setEndpoint("https://eastus.api.cognitive.microsoft.com/")
+      .setInputCol("In - Sentiment")
+      .setOutputCol("Out - Sentiment")
 
-  test("Sentiment Analysis - Basic Usage") {
-    val replies = detector.transform(df)
-      .select("sentiment", "confidenceScores", "sentences", "warnings")
-      .collect()
-    assert(replies(0).getString(0) == "positive" && replies(1).getString(0) == "negative"
-      && replies(2).getString(0) == "neutral")
+
+    test("Sentiment Analysis - Basic Usage") {
+      val outputCol = detector.transform(df)
+        .select("Out - Sentiment")
+        .collect()
+      assert(outputCol(0).schema(0).name == "Out - Sentiment")
+      val fromRow = SentimentResponseV4.makeFromRowConverter
+      outputCol.foreach(row => {
+        val outResponse = fromRow(row.getAs[GenericRowWithSchema]("Out - Sentiment"))
+        assert(outResponse.modelVersion.get == "2020-04-01")

```

Comment:
I suspect this test will fail once we update the sentiment model-version. You might want to change the test to check that the model version is simply not empty, or has a `####-##-##` format.